pkg/autoupdate/tuf/autoupdate.go

```diff
@@ -79,16 +82,28 @@ func WithLogger(logger log.Logger) TufAutoupdaterOption {
 	}
 }
 
+func WithOsqueryRestart(restart func() error) TufAutoupdaterOption {
+	return func(ta *TufAutoupdater) {
+		if ta.restartFuncs == nil {
+			ta.restartFuncs = make(map[autoupdatableBinary]func() error)
+		}
```

Comment:
Nit: I think this can never be nil since were creating in NewTufAutoupdater()

```suggestion
```

Reply:
I think I was being overly cautious in case we accidentally re-ordered `NewTufAutoupdater` to apply opts first, or in case something ends up using the opts without using `NewTufAutoupdater` -- both unlikely, but better safe than sorry?